OMCompiler/SimulationRuntime/OMSICpp/runtime/include/Core/Math/Functions.h

```diff
@@ -35,12 +35,13 @@ inline static int sgn(const double& c)
     return (c < 0) ? -1 : ((c == 0) ? 0 : 1);
 }
 
-/// Definition of Signum function
 double BOOST_EXTENSION_EXPORT_DECL division(const double& a, const double& b, bool throwEx, const char* text);
 
+/// Provides the modulus as defined in modelica (same sign as v2 and abs smaller than abs(v2))
 inline static int modelica_mod_int(int v1, int v2)
 {
-    return v1 % v2;
+    int tmp = v1 % v2;
+    return ((v2 > 0 && tmp < 0) || (v2 < 0 && tmp > 0)) ? (tmp + v2) : tmp;
```

Comment:
All four cases seem to be correct, however, it is very difficult to read.

Reply:
I checked and tested it very carefully, this is the best way I could think of.
I should leave a comment to explain this :sweat_smile: